src/Step.php

```diff
@@ -146,10 +148,27 @@ protected function fmod($x, $y)
         }
 
         //find the maximum precision from both input params to give accurate results
-        $xFloatSegment = substr($x, strpos($x, '.') + 1) ?: '';
-        $yFloatSegment = substr($y, strpos($y, '.') + 1) ?: '';
-        $precision = strlen($xFloatSegment) + strlen($yFloatSegment);
+        $precision = $this->getPrecision($x) + $this->getPrecision($y);
 
         return round($x - $y * floor($x / $y), $precision);
     }
+
+    /**
+     * replaces the internal substraction operation which give wrong results on some cases
+     *
+     * @param float $x
+     * @param float $y
+     * @return float
+     */
+    protected function sub($x, $y)
```

Comment:
should be `private`

Reply:
I choose protected since fmod was in protected also. If I use private for sub, should i use private for fmod as well ?

Reply:
I think `fmod` function should be private also, but that is BC break. Newly introduce internal function should be private imo.

Reply:
I agree, I will make the modification.

Reply:
done